src/Features/Core/Portable/AddImport/AbstractAddImportFeatureService.cs

```diff
@@ -540,5 +541,20 @@ private CodeAction TryCreateCodeAction(Document document, AddImportFixData fixDa
 
             throw ExceptionUtilities.Unreachable;
         }
+
+        protected ITypeSymbol GetAwaitInfo(SemanticModel semanticModel, SyntaxNode node, ISyntaxFactsService syntaxFactsService, CancellationToken cancellationToken)
+        {
+            var awaitExpression = FirstAwaitExpressionAncestor(node, syntaxFactsService);
+
+            var innerExpression = syntaxFactsService.GetExpressionOfAwaitExpression(node);
+
+            return semanticModel.GetTypeInfo(innerExpression).Type;
+        }
+
+        protected bool AncestorOrSelfIsAwaitExpression(SyntaxNode node, ISyntaxFactsService syntaxFactsService)
+            => FirstAwaitExpressionAncestor(node, syntaxFactsService) != null;
+
+        private SyntaxNode FirstAwaitExpressionAncestor(SyntaxNode node, ISyntaxFactsService syntaxFactsService)
+            => node.FirstAncestorOrSelf<SyntaxNode>(n => syntaxFactsService.IsAwaitExpression(n));
```

Comment:
Seems it won't be a problem. We can revisit if the type shows up as relevant during profiling, but since other methods allocate on the same path I'm not expecting a problem.